File: FR_17_04/fr_17_04.py
# https://pl.spoj.com/problems/FR_17_04/

n = int(input())

for _ in range(n):
    Nm, Sm, Ng, Sg = map(int, input().split())
    if Nm*Sm*Sm > Ng*Sg*Sg:
        print("Maciek")
    elif Nm*Sm*Sm == Ng*Sg*Sg:
        print("porcje sa takie same")
    else:
        print("Grzesiek")

File: FR_17_04/fr_17_04.jl
x = parse(Int, readline())

for j in 1:x
    input = split(readline())
    Nm, Sm, Ng, Sg = [parse(Int, input[i]) for i = 1:4]
    Pm = Nm*Sm*Sm
    Pg = Ng*Sg*Sg
    if Pm > Pg 
        println("Maciek")
    elseif Pm == Pg
        println("porcje sa takie same")
    else 
        println("Grzesiek")
    end
end